src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/ConnectivityTests/ConnectivityTest.cs

```diff
@@ -27,14 +27,15 @@ public static class ConnectivityTest
         private static readonly string s_dropDatabaseCmd = $"DROP DATABASE {s_databaseName}";
 
         [ConditionalFact(typeof(DataTestUtility), nameof(DataTestUtility.AreConnStringsSetup))]
-        public static void EnvironmentHostNameTest()
+        public static void EnvironmentHostNameSPIDTest()
         {
             SqlConnectionStringBuilder builder = (new SqlConnectionStringBuilder(DataTestUtility.TCPConnectionString) { Pooling = true });
             builder.ApplicationName = "HostNameTest";
 
             using (SqlConnection sqlConnection = new SqlConnection(builder.ConnectionString))
             {
                 sqlConnection.Open();
+                int sqlclientSPID = sqlConnection.ServerProcessId;
```

Comment:
isn't this read only guaranteed to be consistent with the below query result _after_ the query has completed?

Reply:
Yes I'll add check there too.